src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,108 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class RegistrationServiceImplTest {
+    private static StorageDao storage;
+    private static RegistrationService registration;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        storage = new StorageDaoImpl();
+        registration = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+    }
+
+    @Test
+    void registration_nullUser_NotOk() {
+        assertThrows(RuntimeException.class, () ->
+                registration.register(null)
+        );
+    }
+
+    @Test
+    void registration_userIsAlreadyInStorage_NotOk() {
+        user.setLogin("arangutang1488");
+        storage.add(user);
+        assertThrows(RuntimeException.class, () ->
+                registration.register(user)
+        );
+    }
+
+    @Test
+    void registration_userAgeIsLessThanEighteen_NotOk() {
+        user.setLogin("wrongAge");
+        user.setPassword("wrongAge");
+        user.setAge(-228);
```

Comment:
let's create one user globally and change the parameters of this user that we need for our tests